plugins/@grouparoo/iterable/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,115 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+import { addToList, removeFromList } from "./listMethods";
+
+export const exportProfile: ExportProfilePluginMethod = async (args) => {
+  try {
+    return sendProfile(args);
+  } catch (error) {
+    if (error?.response?.status === 429) {
+      return { error, success: false, retryDelay: 1000 * 11 }; // the most common rate-limit error from hubspot is in 10-second intervals
+    } else {
+      throw error;
+    }
+  }
+};
+
+export const sendProfile: ExportProfilePluginMethod = async ({
+  appGuid,
+  appOptions,
+  destinationOptions,
+  export: {
+    toDelete,
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+  const client = await connect(appOptions);
+  const email = newProfileProperties["email"]; // this is how we will identify profiles
+  const currentEmail = oldProfileProperties["email"];
+  if (!email) {
+    throw new Error(`newProfileProperties[email] is a required mapping`);
+  }
+
+  if (toDelete) {
+    if (currentEmail && currentEmail !== email) {
+      await client.users.delete(currentEmail);
+    } else {
+      await client.users.delete(email);
+    }
+    return { success: true };
+  } else {
+    // create the user and set properties
+    const deletePropertiesPayload = {};
+    const newPropertyKeys = Object.keys(newProfileProperties);
+    Object.keys(oldProfileProperties)
+      .filter((k) => !newPropertyKeys.includes(k))
+      .forEach((k) => (deletePropertiesPayload[k] = null));
+
+    const dataFields = Object.assign(
+      {},
+      deletePropertiesPayload,
+      newProfileProperties
+    );
+    const formattedDataFields = {};
+    for (const key of Object.keys(dataFields)) {
+      if (key !== "email") {
+        formattedDataFields[key] = formatVar(dataFields[key]);
+      }
+    }
+    const payload = Object.assign(
+      { email },
+      { dataFields: formattedDataFields }
+    );
+
+    if (currentEmail && currentEmail !== email) {
+      const oldUser = await getUser(client, currentEmail);
+      if (oldUser) {
+        const emailPayload = { currentEmail, newEmail: email };
+        if (newProfileProperties.userId) {
+          emailPayload["currentUserId"] = newProfileProperties.userId;
+        }
+        await client.users.updateEmail(emailPayload);
+      }
+    }
+
+    await client.users.update(payload);
+    const currentUser = await getUser(client, email);
```

Comment:
When this user is added, don't we have the issue of it taking N seconds. I think if you had a test that added a new user _and_ put them groups we might have a problem here in that it would be null. Or if you changed the email and added groups, it wouldn't find them here.

The code I suggested before had this above `if (currentEmail && currentEmail !== email)` and reassigned the `currentUser` to the changed one.
```
  let currentUser = await client.users.get(email);
  if (currentEmail && email !== currentEmail) {
    const oldUser = await client.users.get(oldEmail);
    if (oldUser) {
      ....
      await client.users.updateEmail(emailPayload);
      currentUser = oldUser;
      currentUser.email = email;
    }
  }
```

With that approach a null `currentUser` will be known to not be in any groups. and you won't have that `indexContacts` type issue.